scripts/core/editor3/components/Editor3.jsx

```diff
@@ -0,0 +1,226 @@
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Editor, EditorState, RichUtils, ContentState} from 'draft-js';
+
+export class Editor3 extends React.Component {
+    /**
+     * @ngdoc react component
+     * @module superdesk.core.editor3
+     * @name Editor3
+     * @param {Array} editorFormat the formating settings available for editor
+     * @param {String} value the model for editor value
+     * @param {Boolean} readOnly if true the editor is read only
+     * @param {Boolean} showToolbar if true the editor will show the toolbar
+     * @param {Function} onChange the callback executed when the editor value is changed
+     * @param {String} language the current language used for spellchecker
+     * @description Editor3 is a draft.js based editor that support customizable formatting (shortly more features will come ...)
+     */
+    constructor(props) {
+        super(props);
+        var initialContentState = ContentState.createFromText(props.value);
+        this.state = {editorState: EditorState.createWithContent(initialContentState)};
+
+        this.readOnly = props.readOnly || false;
+        this.showToolbar = props.showToolbar || false;
+        this.editorFormat = props.editorFormat || [];
+        this.parentOnChange = props.onChange;
+
+        this.focus = this.focus.bind();
+        this.onChange = this.onChange.bind(this);
+
+        this.handleKeyCommand = this.handleKeyCommand.bind(this);
+        this.onTab = this.onTab.bind(this);
+        this.toggleBlockType = this.toggleBlockType.bind(this);
+        this.toggleInlineStyle = this.toggleInlineStyle.bind(this);
+    }
+
+    focus() {
+        this.refs.editor.focus();
+    }
+
+    onChange(editorState) {
+        this.setState({editorState});
+        this.parentOnChange(editorState.getCurrentContent().getPlainText());
+    }
+
+    handleKeyCommand(command) {
```

Comment:
Please document all public functions (in JS the pattern is that functions which are not public start with underscore `_`). In this case, all of the methods are public and they need to be documented.